pages/product.js

```diff
@@ -1,53 +1,58 @@
+const PriceHelper = require('../helpers/priceHelper');
+const priceHelper = new PriceHelper();
+const ElementCheckingHelper = require('../helpers/elementCheckingHelper');
+const elCheck = new ElementCheckingHelper();
 const { I } = inject();
-
-
 module.exports = {
     colorDropDown: { xpath: '//label[text()="Color"]/following-sibling::div/a[1]' },
     sizeDropDown: { xpath: '//label[text()="Size"]/following-sibling::div/a[1]' },
     colorOption: { xpath: '//label[text()="Color"]/following-sibling::div/ul/li[2]/a' },
     sizeOption: { xpath: '//label[text()="Size"]/following-sibling::div/ul/li[2]/a' },
     productPriceText: { xpath: '//*[@id="content"]/div[1]/div[2]/div/div[1]/span' },
+    available: { xpath: '//*[@id="content"]/div[1]/div[2]/div/ul/li[last()]' },
 
-     parsePrice(priceString) {
-        return parseFloat(priceString.replace(/[^0-9.-]/g, ''));
+    async checkAvailable() {
+        if (await I.grabTextFrom(this.available) == 'Availability: In Stock') {
+            console.log('Product is available')
+        } else {
+            throw new Error('Product is n/a');
+        }
     },
 
-    selectColor() {
-        I.click(this.colorDropDown);
-        I.click(this.colorOption);
+    async checkSizeExists() {
+        return elCheck.checkElementExists(this.sizeDropDown);
     },
 
-    selectSize() {
-        I.click(this.sizeDropDown);
-        I.click(this.sizeOption);
+    async checkColorExists() {
+        return elCheck.checkElementExists(this.colorDropDown);
     },
 
-    async checkSizeExists() {
-        return await tryTo(() => I.seeElement(this.sizeDropDown));
+    async selectColor() {
+        if (await this.checkColorExists() == true) {
```

Comment:
use `await elCheck.checkElementExists(this.colorDropDown)` in condition, then you can remove `checkColorExists()` method.
Same for size.